banyan/resource_api_key.go

```diff
@@ -39,82 +37,86 @@ func resourceApiKey() *schema.Resource {
 			"secret": {
 				Type:        schema.TypeString,
 				Description: "API Secret key",
-				Computed:    true,
 				Sensitive:   true,
+				Computed:    true,
 				ForceNew:    true,
 			},
 			"scope": {
-				Type:        schema.TypeString,
-				Description: "API Secret key",
-				Optional:    true,
+				Type:         schema.TypeString,
+				Description:  "Scope for the API key. Must be one of: \"satellite\", \"access_tier\", \"read_logs\", \"Admin\", \"ServiceAuthor\", \"PolicyAuthor\", \"EventWriter\", \"ReadOnly\"",
```

Comment:
Is this a more generic API key description or specific to access tier? If it's for access tier, then it needs to be `access_tier` or `Admin` scope.

Reply:
This is a generic resource for the API key so you can make any API key type